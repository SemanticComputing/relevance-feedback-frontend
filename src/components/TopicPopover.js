import React from 'react';
import { bool, object, string } from 'prop-types';
import { Popover, PopoverHeader, PopoverBody } from 'reactstrap';
import TopicWords from './TopicWords';

const TopicPopover = ({ isOpen, topic, target }) => {

  return (
    <Popover isOpen={isOpen} target={`[id="${target}"]`} placement={'right'}>
      <PopoverHeader>Topiikki {topic.topic}</PopoverHeader>
      <PopoverBody>
        <TopicWords topic={topic.topicWords} />
      </PopoverBody>
    </Popover>
  );
};

TopicPopover.propTypes = {
  isOpen: bool,
  topic: object.isRequired,
  target: string.isRequired
};

export default TopicPopover;
