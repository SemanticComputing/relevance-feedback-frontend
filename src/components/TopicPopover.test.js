import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import TopicPopover from './TopicPopover';
import * as reactstrap from 'reactstrap';

const topic = {
  topicWords: [
    ['word', 0.2],
    ['something', 0.1489],
    ['innovation', 0.1],
    ['trash', 0.01]
  ],
  topic: 0
};

const target = 'http://example.net';

// These tests are perhaps not great as Popover is mocked due to difficulties
// in getting it to render in a test environment.

// Make Popover return its child components
const popoverMock = jest.fn((stuff) => stuff.children);
reactstrap.Popover = popoverMock;  // eslint-disable-line import/namespace

describe('TopicPopover', () => {

  beforeEach(() => {
    popoverMock.mockClear();
  });

  it('renders without crashing', () => {
    const isOpen = true;

    const div = document.createElement('div');
    div.setAttribute('id', target);
    document.body.appendChild(div);
    ReactDOM.render(<TopicPopover target={target} isOpen={isOpen} topic={topic} currentTopic={{}} />, div);
    document.body.removeChild(div);
  });

  it('renders a topic', () => {
    const isOpen = true;

    const tree = renderer
      .create(<TopicPopover target={target} isOpen={isOpen} topic={topic} currentTopic={{}} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should pass isOpen to Popover', () => {
    mount(<TopicPopover target={target} isOpen={true} topic={topic} currentTopic={{}} />);

    expect(popoverMock.mock.calls.length).toBe(1);
    expect(popoverMock.mock.calls[0][0].isOpen).toBe(true);

    popoverMock.mockClear();

    mount(<TopicPopover target={target} isOpen={false} topic={topic} currentTopic={{}} />);

    expect(popoverMock.mock.calls.length).toBe(1);
    expect(popoverMock.mock.calls[0][0].isOpen).toBe(false);
  });
});
