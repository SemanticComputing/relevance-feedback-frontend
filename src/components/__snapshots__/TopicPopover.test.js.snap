// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicPopover renders a topic 1`] = `
Array [
  <h3
    className="popover-header"
>
    Topiikki 
    0
</h3>,
  <div
    className="popover-body"
>
    <div>
        <span
            style={
                Object {
                    "fontSize": 260,
                  }
            }
        >
            word
             
        </span>
        <span
            style={
                Object {
                    "fontSize": 193.57,
                  }
            }
        >
            something
             
        </span>
        <span
            style={
                Object {
                    "fontSize": 130,
                  }
            }
        >
            innovation
             
        </span>
        <span
            style={
                Object {
                    "fontSize": 13,
                  }
            }
        >
            trash
             
        </span>
    </div>
</div>,
]
`;
